File: 515-find-largest-value-in-each-tree-row/find-largest-value-in-each-tree-row.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def largestValues(self, root: Optional[TreeNode]) -> List[int]:
        ans = []
        if not root:
            return []
        result = []
        queue = [root]
        while len(queue) > 0 :
            row = []
            rowSize = len(queue)

            for i in range(rowSize):
                node = queue.pop(0)
                row.append(node.val)

                if node.left :
                    queue.append(node.left)
                if node.right :
                    queue.append(node.right)
            result.append(row)
        for i in range(len(result)):
            arr = result[i]
            maxVal = max(arr)
            ans.append(maxVal)
        return ans

File: 515-find-largest-value-in-each-tree-row/find-largest-value-in-each-tree-row.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var largestValues = function (root) {
    if (!root) return []; 
    const result = []; 
    const queue = [root]; 
    let ans = []
    while (queue.length > 0) {
        const row = []; 
        const rowSize = queue.length; 

        for (let i = 0; i < rowSize; i++) {
            const node = queue.shift();
            row.push(node.val);
            if (node.left) queue.push(node.left);
            if (node.right) queue.push(node.right);
        }

        result.push(row);
    }
    console.log(result)
    for(let i = 0 ; i < result.length ; i++){
        const arr = result[i]
        const max = Math.max(...arr)
        ans.push(max)
    }
    return ans
};